### ISSUE TITLE
Track modelimport challenges

### ISSUE THREAD

**Issue by tomthetrainer** [deeplearning4j/deeplearning4j#2570]
Creating an issue to track progress of importing important keras models using model import

**Comment by tomthetrainer** [comment]
inception_v3 

Progress: 
Was able to load config

Unable to load weights

Config created with 
'''
ComputationGraphConfiguration model = KerasModelImport.importKerasModelConfiguration("inception_V3_config");
'''
inception_V3_config was created with, 
'''
json_string = model.to_json()
    text_file = open("inception_V3_config", "w")
    text_file.write(json_string)
    text_file.close()
'''

required keras.json 
'''
{"image_dim_ordering": "th","epsilon": 1e-07, "floatx": "float32", "backend": "theano"}
'''

Used model grabbed from here
https://github.com/fchollet/deep-learning-models

*note those only worked with python 2.7*
Python 2.7.12

Keras version 
1.2.0

Error log when attempting to load weights
'''
o.n.l.f.Nd4jBackend - Loaded [CpuBackend] backend
Exception in thread "main" java.lang.ExceptionInInitializerError
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.<init>(NativeOpExecutioner.java:38)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:5646)
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:5554)
	at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:189)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.readWeightsFromHdf5(KerasModelImport.java:448)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.<init>(KerasModelImport.java:330)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:185)
	at org.deeplearning4j.examples.recurrent.basic.Inception_V3_from_Keras.main(Inception_V3_from_Keras.java:29)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.lang.RuntimeException: ND4J is probably missing dependencies. For more information, please refer to: http://nd4j.org/getstarted.html
	at org.nd4j.nativeblas.NativeOpsHolder.<init>(NativeOpsHolder.java:40)
	at org.nd4j.nativeblas.NativeOpsHolder.<clinit>(NativeOpsHolder.java:16)
	... 18 more
Caused by: java.lang.NullPointerException
	at org.nd4j.nativeblas.NativeOpsHolder.<init>(NativeOpsHolder.java:24)
	... 19 more
'''

**Comment by tomthetrainer** [comment]
VGG16, 

Same as above, config loads weights fail

**Comment by tomthetrainer** [comment]
resnet

Fails to load config, 
'''
Exception in thread "main" org.deeplearning4j.nn.modelimport.keras.InvalidKerasConfigurationException: Unsupported keras layer type ZeroPadding2D. For more information, see http://deeplearning4j.org/model-import-keras.
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.buildLayerFromConfig(KerasLayer.java:468)
'''

**Comment by eraly** [comment]
To get around the zero padding issue, I used the config I wrote out from keras after loading from this:
https://github.com/fchollet/deep-learning-models/blob/master/vgg16.py

The zero padding is not necessary and the config from this script will not output that. Here is the config I wrote out. Why won't it let me attach it :(

```
{"class_name": "Model", "keras_version": "1.2.0", "config": {"layers": [{"class_name": "InputLayer", "config": {"batch_input_shape": [null, 3, 224, 224], "input_dtype": "float32", "sparse": false, "name": "input_2"}, "inbound_nodes": [], "name": "input_2"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block1_conv1", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 64, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["input_2", 0, 0]]], "name": "block1_conv1"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block1_conv2", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 64, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block1_conv1", 0, 0]]], "name": "block1_conv2"}, {"class_name": "MaxPooling2D", "config": {"name": "block1_pool", "trainable": true, "dim_ordering": "th", "pool_size": [2, 2], "strides": [2, 2], "border_mode": "valid"}, "inbound_nodes": [[["block1_conv2", 0, 0]]], "name": "block1_pool"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block2_conv1", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 128, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block1_pool", 0, 0]]], "name": "block2_conv1"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block2_conv2", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 128, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block2_conv1", 0, 0]]], "name": "block2_conv2"}, {"class_name": "MaxPooling2D", "config": {"name": "block2_pool", "trainable": true, "dim_ordering": "th", "pool_size": [2, 2], "strides": [2, 2], "border_mode": "valid"}, "inbound_nodes": [[["block2_conv2", 0, 0]]], "name": "block2_pool"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block3_conv1", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 256, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block2_pool", 0, 0]]], "name": "block3_conv1"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block3_conv2", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 256, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block3_conv1", 0, 0]]], "name": "block3_conv2"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block3_conv3", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 256, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block3_conv2", 0, 0]]], "name": "block3_conv3"}, {"class_name": "MaxPooling2D", "config": {"name": "block3_pool", "trainable": true, "dim_ordering": "th", "pool_size": [2, 2], "strides": [2, 2], "border_mode": "valid"}, "inbound_nodes": [[["block3_conv3", 0, 0]]], "name": "block3_pool"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block4_conv1", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block3_pool", 0, 0]]], "name": "block4_conv1"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block4_conv2", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block4_conv1", 0, 0]]], "name": "block4_conv2"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block4_conv3", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block4_conv2", 0, 0]]], "name": "block4_conv3"}, {"class_name": "MaxPooling2D", "config": {"name": "block4_pool", "trainable": true, "dim_ordering": "th", "pool_size": [2, 2], "strides": [2, 2], "border_mode": "valid"}, "inbound_nodes": [[["block4_conv3", 0, 0]]], "name": "block4_pool"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block5_conv1", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block4_pool", 0, 0]]], "name": "block5_conv1"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block5_conv2", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block5_conv1", 0, 0]]], "name": "block5_conv2"}, {"class_name": "Convolution2D", "config": {"W_constraint": null, "b_constraint": null, "name": "block5_conv3", "activity_regularizer": null, "trainable": true, "dim_ordering": "th", "nb_col": 3, "subsample": [1, 1], "init": "glorot_uniform", "bias": true, "nb_filter": 512, "border_mode": "same", "b_regularizer": null, "W_regularizer": null, "activation": "relu", "nb_row": 3}, "inbound_nodes": [[["block5_conv2", 0, 0]]], "name": "block5_conv3"}, {"class_name": "MaxPooling2D", "config": {"name": "block5_pool", "trainable": true, "dim_ordering": "th", "pool_size": [2, 2], "strides": [2, 2], "border_mode": "valid"}, "inbound_nodes": [[["block5_conv3", 0, 0]]], "name": "block5_pool"}, {"class_name": "Flatten", "config": {"trainable": true, "name": "flatten"}, "inbound_nodes": [[["block5_pool", 0, 0]]], "name": "flatten"}, {"class_name": "Dense", "config": {"W_constraint": null, "b_constraint": null, "name": "fc1", "activity_regularizer": null, "trainable": true, "init": "glorot_uniform", "bias": true, "input_dim": 25088, "b_regularizer": null, "W_regularizer": null, "activation": "relu", "output_dim": 4096}, "inbound_nodes": [[["flatten", 0, 0]]], "name": "fc1"}, {"class_name": "Dense", "config": {"W_constraint": null, "b_constraint": null, "name": "fc2", "activity_regularizer": null, "trainable": true, "init": "glorot_uniform", "bias": true, "input_dim": 4096, "b_regularizer": null, "W_regularizer": null, "activation": "relu", "output_dim": 4096}, "inbound_nodes": [[["fc1", 0, 0]]], "name": "fc2"}, {"class_name": "Dense", "config": {"W_constraint": null, "b_constraint": null, "name": "predictions", "activity_regularizer": null, "trainable": true, "init": "glorot_uniform", "bias": true, "input_dim": 4096, "b_regularizer": null, "W_regularizer": null, "activation": "softmax", "output_dim": 1000}, "inbound_nodes": [[["fc2", 0, 0]]], "name": "predictions"}], "input_layers": [["input_2", 0, 0]], "output_layers": [["predictions", 0, 0]], "name": "model_2"}}

```

**Comment by tomthetrainer** [comment]
closing, out of date
